pkg/apis/testharness/v1beta1/test_types.go

```diff
@@ -132,5 +137,58 @@ type Command struct {
 	SkipLogOutput bool `json:"skipLogOutput"`
 }
 
+// TestCollector are post assert / error commands that allow for the collection of information sent to the test log
+type TestCollector struct {
+	// The pod name to access logs.
+	Pod string `json:"pod"`
+	// namespace to use.
+	Namespace string `json:"namespace"`
+	// Container in pod to get logs from
```

Comment:
Might be worth mentioning here that container name is required if the pod contains more than one of them (since the `kubectl logs` fails in that case).

Alternatively, I think it might be a nice usability feature if we instead collected all containers' (including init ones which might have been injected at runtime) logs by default. This seems easy to do these days by just appending `--all-containers`.

Reply:
@porridge agree!   `--all-containers` added if no container specified.